docs/design/features/globalization-icu.md

```diff
@@ -0,0 +1,131 @@
+# .NET Globalization ICU on all runtimes
+
+Author: [Santiago Fernandez Madero](https://github.com/safern)
+
+In .NET Core our globalization APIs behavior used different underlying libraries, in Unix, we used [International Components for Unicode (ICU)](http://site.icu-project.org/home) and in Windows, we used [National Language Support (NLS)](https://docs.microsoft.com/en-us/windows/win32/intl/national-language-support). This caused some behavior differences in a handful number of Globalization APIs:
+
+- Cultures and culture data
+- String casing
+- String sorting and searching
+- Sort keys
+- String Normalization
+- Internationalized Domain Names (IDN) support
+- Time Zone display name on Linux
+
+## ICU on Windows
+
+Windows 10 May 2019 Update and later versions started including [icu.dll](https://docs.microsoft.com/en-us/windows/win32/intl/international-components-for-unicode--icu-) as part of the OS, so in .NET 5.0 we changed the default behavior for Globalization APIs when running in Windows, which tries to load `icu.dll` and if we can load it, we use ICU as the underlying native APIs, if we fail to load it, we use the legacy behavior (NLS), this will be the case for older versions of Windows.
+
+*Note: CurrentCulture, CurrentUICulture and CurrentRegion still use Windows OS APIs to honor user settings*
```

Comment:
> *Note: CurrentCulture, CurrentUICulture and CurrentRegion still use Windows OS APIs to honor user settings*

Can you clarify this? Do you mean when using those in some API, we'll use NLS instead (that seems quite limiting, but if that's the design fine)? Or do you mean we will ask Windows for the culture/region to use, but then use ICU tables to do the job.

Reply:
Those are user profile settings on Windows. I would rephrase this as `CurrentCulture, CurrentUICulture and CurrentRegion properties are still initialized the same way we used to initialize using the Windows Win32 APIs and not ICU APIs`

Reply:
But the point being that if I use ICU (possibly my own ICU) then yes I do get ICU behavior for `string.Contains(... , CurrentCulture)` etc?

Reply:
Yeah, it is just the initialization routines to get the current culture info from the OS that uses NLS, but for all other operations you would use ICU.

Reply:
that is right. the functionality is fully provided by using ICU. initializing current culture is mainly reading the culture name from the OS settings and will use ICU to initialize the culture and performing any operations with it.